src/ATokenNaive.sol

```diff
@@ -0,0 +1,357 @@
+    _writeCheckpoint(_checkpoints[from], _subtractionFn, amount);
+    _writeCheckpoint(_totalDepositCheckpoints, _subtractionFn, amount);
+    // End modifications.
+
+    _burnScaled(from, receiverOfUnderlying, amount, index);
+    if (receiverOfUnderlying != address(this)) {
+      IERC20(_underlyingAsset).safeTransfer(receiverOfUnderlying, amount);
+    }
+  }
+
+  /// Note: this has been modified from Aave v3's ScaledBalanceTokenBase
+  /// contract to include balance checkpointing code.
+  /// https://github.com/aave/aave-v3-core/blob/f3e037b3638e3b7c98f0c09c56c5efde54f7c5d2/contracts/protocol/tokenization/base/ScaledBalanceTokenBase.sol#L61-L91
+  /// Modifications are as indicated below.
+  ///
+  /// @notice Implements the basic logic to mint a scaled balance token.
+  /// @param caller The address performing the mint
+  /// @param onBehalfOf The address of the user that will receive the scaled tokens
+  /// @param amount The amount of tokens getting minted
+  /// @param index The next liquidity index of the reserve
+  /// @return `true` if the the previous balance of the user was 0
+  function __mintScaled(
+    address caller,
+    address onBehalfOf,
+    uint256 amount,
+    uint256 index
+  ) internal returns (bool) {
+    uint256 amountScaled = amount.rayDiv(index);
+    require(amountScaled != 0, Errors.INVALID_MINT_AMOUNT);
+
+    uint256 scaledBalance = super.balanceOf(onBehalfOf);
+    uint256 balanceIncrease = scaledBalance.rayMul(index) -
+      scaledBalance.rayMul(_userState[onBehalfOf].additionalData);
+
+    _userState[onBehalfOf].additionalData = index.toUint128();
+
+    _mint(onBehalfOf, amountScaled.toUint128());
+
+    // Begin modifications.
+    // We increment by `amount` instead of any computed/rebased amounts because
+    // `amount` is what actually gets transferred of the underlying asset. We
+    // need our checkpoints to still match up with the underlying asset balance.
+    deposits[onBehalfOf] += amount;
+    _writeCheckpoint(_checkpoints[onBehalfOf], _additionFn, amount);
+    _writeCheckpoint(_totalDepositCheckpoints, _additionFn, amount);
+    // End modifications.
+
+    uint256 amountToMint = amount + balanceIncrease;
+    emit Transfer(address(0), onBehalfOf, amountToMint);
+    emit Mint(caller, onBehalfOf, amountToMint, balanceIncrease, index);
+
+    return (scaledBalance == 0);
+  }
+  //===========================================================================
+  // END: Aave overrides
+  //===========================================================================
+
+  //===========================================================================
+  // BEGIN: Checkpointing code.
+  //===========================================================================
+  // This was been copied from OZ's ERC20Votes checkpointing system with minor
+  // revisions:
+  //   * Replace "Vote" with "Deposit", as deposits are what we need to track
+  //   * Make some variable names longer for readibility
```

Comment:
```suggestion
  //   * Make some variable names longer for readability
```

Also which ones were make longer? From a quick glance at ERC20Votes I couldn't tell and they seem the same

Reply:
I think I made some variables in the binary search a little longer when I was reading it to help me understand it